tests/unit/test__job_helpers.py

```diff
@@ -0,0 +1,314 @@
+    response = {
+        "jobReference": {
+            "projectId": project_id,
+            "jobId": job_id,
+            "location": location,
+        },
+        "jobComplete": completed,
+    }
+    if errors is not None:
+        response["errors"] = errors
+    return response
+
+
+@pytest.mark.parametrize(
+    ("job_config", "expected"),
+    (
+        (None, make_query_request()),
+        (QueryJobConfig(), make_query_request()),
+        (
+            QueryJobConfig(default_dataset="my-project.my_dataset"),
+            make_query_request(
+                {
+                    "defaultDataset": {
+                        "projectId": "my-project",
+                        "datasetId": "my_dataset",
+                    }
+                }
+            ),
+        ),
+        (QueryJobConfig(dry_run=True), make_query_request({"dryRun": True})),
+        (
+            QueryJobConfig(use_query_cache=False),
+            make_query_request({"useQueryCache": False}),
+        ),
+        (
+            QueryJobConfig(use_legacy_sql=True),
+            make_query_request({"useLegacySql": True}),
+        ),
+        (
+            QueryJobConfig(
+                query_parameters=[
+                    ScalarQueryParameter("named_param1", "STRING", "param-value"),
+                    ScalarQueryParameter("named_param2", "INT64", 123),
+                ]
+            ),
+            make_query_request(
+                {
+                    "parameterMode": "NAMED",
+                    "queryParameters": [
+                        {
+                            "name": "named_param1",
+                            "parameterType": {"type": "STRING"},
+                            "parameterValue": {"value": "param-value"},
+                        },
+                        {
+                            "name": "named_param2",
+                            "parameterType": {"type": "INT64"},
+                            "parameterValue": {"value": "123"},
+                        },
+                    ],
+                }
+            ),
+        ),
+        (
+            QueryJobConfig(
+                query_parameters=[
+                    ScalarQueryParameter(None, "STRING", "param-value"),
+                    ScalarQueryParameter(None, "INT64", 123),
+                ]
+            ),
+            make_query_request(
+                {
+                    "parameterMode": "POSITIONAL",
+                    "queryParameters": [
+                        {
+                            "parameterType": {"type": "STRING"},
+                            "parameterValue": {"value": "param-value"},
+                        },
+                        {
+                            "parameterType": {"type": "INT64"},
+                            "parameterValue": {"value": "123"},
+                        },
+                    ],
+                }
+            ),
+        ),
+        # TODO: connection properties
```

Comment:
In this or in a separate PR?

Reply:
Oops this PR. I created https://github.com/googleapis/python-bigquery/pull/1024 shortly after writing this TODO.